katdal/h5datav3.py

```diff
@@ -596,7 +596,7 @@ def _get_telstate_attr(self, key, default=None, no_unpickle=()):
             # the most recent value.
             try:
                 return self.sensor['TelescopeState/' + key][-1]
-            except (KeyError, IndexError):
+            except (KeyError, IndexError, TypeError):
```

Comment:
Why would this give a `TypeError` rather than an `AttributeError`?

Either way, I think a comment would be useful to make the reader aware that the case of the sensor cache not existing yet needs to be handled. Otherwise someone in future might not see where the exception could come from and remove it again.

For completeness, an alternative would be initially set `self.sensor = {}` so that the sensor cache appears empty rather than missing. I'm not sure I like that though, since the type changes when the real cache is created.

Reply:
It's a `TypeError` because I set `self.sensor` to `None` at the start.

I like your `dict` suggestion. It is not that far-fetched since `SensorCache` is a derived class of `dict`... Alternatively I could cobble together an empty `SensorCache`.